tasks/main.yml

```diff
@@ -78,23 +75,23 @@
     owner: root
     group: root
     mode: 0644
-  # Prevent Ansible from displaying password
   no_log: true
   notify: restart haproxy_exporter
 
-- name: ensure haproxy_exporter is enabled
+- name: start and enable haproxy_exporter service
   become: true
   systemd:
     name: haproxy_exporter
     enabled: "{{ haproxy_exporter_service_enabled }}"
     state: "{{ haproxy_exporter_service_state }}"
+    daemon_reload: true
 
 - name: curl haproxy-exporter metrics
   uri:
     url: http://localhost:9101/metrics
     return_content: yes
   register: metrics
-  check_mode: no
+  changed_when: false
```

Comment:
I don't believe this is necessary, e.g. our Keycloak tests don't include and don't show changes on each run:

```
- name: test that keycloak login page loads
  uri:
    url: "https://{{ inventory_hostname }}:8443/auth/"
    return_content: yes
    timeout: 60
  register: login_page
  failed_when: "'Welcome to Keycloak' not in login_page.content"
```

(also looks like we can combine the following task into this one)

Reply:
Removed